server/models/Group.js

```diff
@@ -223,6 +223,22 @@ module.exports = function(Sequelize, DataTypes) {
         });
       },
 
+      findOrAddUserWithRole(user, role) {
```

Comment:
It's used in the new flow. `findAndAddUser` seems better name and I think you are right, it should return a User. I'll make the change.